massa-execution-worker/src/speculative_async_pool.rs

```diff
@@ -287,7 +287,7 @@ impl SpeculativeAsyncPool {
                 }
                 Present(SetUpdateOrDelete::Update(msg_update)) => {
                     current_changes.entry(message_id).and_modify(|e| {
-                        e.apply(msg_update.clone());
+                        *e = msg_update.clone();
                     });
                     return true;
                 }
```

Comment:
question to make sure: do we properly handle the case where there is only an update in the speculative part, and the message is in the final state ?

Reply:
Seems like it to me yes, in that case:
- On line 264 we find an Update in the pool
- Then, we find nothing in the active history so we do nothing (at line 294)
- Then, we find the message in the final state and we apply the update.
@sydhds do you agree?